src/Utils/MX/Methods/LoadRandomMap.as

```diff
@@ -226,10 +226,28 @@ namespace MX
                 return;
             }
             RandomMapIsLoading = true;
-            if (preloadedMap is null && !isLoadingPreload) PreloadRandomMap();
-            while (isLoadingPreload) yield();
-            MX::MapInfo@ map = preloadedMap;
-            @preloadedMap = null;
+            MX::MapInfo@ map;
+            if (RMC::ContinueSavedRun && !RMC::IsInited) {
+                string url = "https://trackmania.exchange/api/maps/get_map_info/id/" + tostring(RMC::CurrentMapID);
```

Comment:
Please use `MX_URL` constant

Reply:
Iirc this endpoint does not exist on https://map-monitor.xk.io/

Reply:
Then add a condition that uses `MX_URL` on MP4 and trackmania.exchange on TMNEXT

Reply:
alright, also going to make an issue on the map-monitor repository for support of the endpoint